components/tidb_query/src/codec/row/v2/row_slice.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+#![allow(dead_code)]
+use crate::codec::{Error, Result};
+use codec::prelude::*;
+use num_traits::PrimInt;
+use tikv_util::codec::read_slice;
+
+#[derive(Default, Debug)]
+struct RowSlice<'a> {
```

Comment:
I don't think it is good to derive `Default` for it, because the default `RowSlice` is very likely to be in a broken state so that removing the `Default` will avoid abusing it elsewhere.

Reply:
Without derived `Default`, in `from_bytes` I need to declare all fields of  `RowSlice`  as variables, then pass it to  `RowSlice`, what's your opinion?
```
    fn from_bytes(mut data: &[u8]) -> Result<RowSlice> {
        assert_eq!(data.read_u8()?, super::CODEC_VERSION);
        //        let mut row = RowSlice::default();
        let is_big = data.read_u8()? == super::Flags::BIG.bits();

        // read ids count
        let non_null_cnt = data.read_u16_le()? as usize;
        let null_cnt = data.read_u16_le()? as usize;
        if is_big {
            let non_null_ids_big = read_ints_le(&mut data, non_null_cnt)?;
            let null_ids_big = read_ints_le(&mut data, null_cnt)?;
            let offsets_big = read_ints_le(&mut data, non_null_cnt)?;
            Ok(RowSlice {
                is_big,
                non_null_ids: &[],
                null_ids: &[],
                offsets: &[],
                offsets_big,
                null_ids_big,
                non_null_ids_big,
                values: data,
            })
        } else {
            let non_null_ids = read_ints_le(&mut data, non_null_cnt)?;
            let null_ids = read_ints_le(&mut data, null_cnt)?;
            let offsets = read_ints_le(&mut data, non_null_cnt)?;
            Ok(RowSlice {
                is_big,
                non_null_ids,
                null_ids,
                offsets,
                offsets_big: &[],
                null_ids_big: &[],
                non_null_ids_big: &[],
                values: data,
            })
        }
```

Reply:
Looks fine to me. Maybe a less duplication way can be...

```rust
let mut non_null_ids = &[];
let mut null_ids = &[];
...
if is_big {
    non_null_ids = ...
} else {
...
}

Ok(RowSlice {
    is_big,
    non_null_ids,
    null_ids,
    ..
})
```

Reply:
done